src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -4676,54 +4597,36 @@ protected sealed override TypeInfo GetTypeInfoCore(SyntaxNode node, Cancellation
 
         protected sealed override IAliasSymbol GetAliasInfoCore(SyntaxNode node, CancellationToken cancellationToken)
         {
-            var nameSyntax = node as IdentifierNameSyntax;
-            return nameSyntax == null ? null : GetAliasInfo(nameSyntax, cancellationToken);
+            return node is IdentifierNameSyntax nameSyntax ? GetAliasInfo(nameSyntax, cancellationToken) : null;
         }
 
         protected sealed override PreprocessingSymbolInfo GetPreprocessingSymbolInfoCore(SyntaxNode node)
         {
-            var nameSyntax = node as IdentifierNameSyntax;
-            return nameSyntax == null ? PreprocessingSymbolInfo.None : GetPreprocessingSymbolInfo(nameSyntax);
+            return node is IdentifierNameSyntax nameSyntax
+                ? GetPreprocessingSymbolInfo(nameSyntax)
+                : PreprocessingSymbolInfo.None;
         }
 
-        protected sealed override ISymbol GetDeclaredSymbolCore(SyntaxNode declaration, CancellationToken cancellationToken)
+        protected sealed override ISymbol GetDeclaredSymbolCore(SyntaxNode node, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
-            var node = (CSharpSyntaxNode)declaration;
-
-            var accessor = node as AccessorDeclarationSyntax;
```

Comment:
AccessorDeclarationSyntax is sealed, so I think it should be moved to the next switch (on kind).